src/main/kotlin/uk/gov/justice/digital/hmpps/approvedpremisesapi/jpa/entity/TaskEntity.kt

```diff
@@ -29,6 +29,15 @@ interface TaskRepository : JpaRepository<Task, UUID> {
       )
     """
 
+    private const val CRN_OR_NAME_QUERY = """
+      :crnOrName IS NULL OR 
+      (
+          ((SELECT COUNT(1) FROM applications a WHERE a.id = application.id AND a.crn = UPPER(:crnOrName)) = 1)
+          OR
+          ((SELECT COUNT(1) FROM approved_premises_applications apa WHERE apa.id = application.id AND apa.name LIKE UPPER('%' || :crnOrName || '%')) = 1)
```

Comment:
would it be more efficient to ensure all three queries have a join on 'application' and 'approved_premises_application' and then reference here without the need for an inline query?

Reply:
Yeah, I nabbed this query from elsewhere without really thinking - have sorted 👍